src/worktree.rs

```diff
@@ -0,0 +1,435 @@
+use crate::repository::Repository;
+use crate::util::path_to_string;
+use napi::bindgen_prelude::{Reference, SharedReference};
+use napi::Env;
+use napi_derive::napi;
+use std::ops::Deref;
+use std::path::Path;
+
+#[napi(object)]
+/// Lock Status of a worktree
+pub struct WorktreeLockStatus {
+  /// Worktree is Unlocked
+  pub status: WorktreeLockStatusType,
+  /// Worktree is locked with the optional message
+  pub reason: Option<String>,
+}
+// (Option<String>)
+#[napi(string_enum)]
+/// Lock Status of a worktree
+pub enum WorktreeLockStatusType {
+  /// Worktree is Unlocked
+  Unlocked,
+  /// Worktree is locked with the optional message
+  Locked,
+}
+
+#[napi(object)]
+/// Options for adding a worktree.
+pub struct WorktreeAddOptions {
+  /// If enabled, this will cause the newly added worktree to be locked.
+  ///
+  /// Defaults to `false`.
+  pub lock: Option<bool>,
+
+  /// If enabled, this will checkout the existing branch matching the worktree name.
+  ///
+  /// Defaults to `false`.
+  pub checkout_existing: Option<bool>,
+
+  /// reference name to use for the new worktree HEAD
+  ///
+  /// Defaults to `null`.
+  pub ref_name: Option<String>,
+}
+
+#[napi(object)]
+/// Options to configure how worktree pruning is performed.
+pub struct WorktreePruneOptions {
+  /// Controls whether valid (still existing on the filesystem) worktrees will be pruned.
+  ///
+  /// Defaults to `false`.
+  pub valid: Option<bool>,
+
+  /// Controls whether locked worktrees will be pruned.
+  ///
+  /// Defaults to `false`.
+  pub locked: Option<bool>,
+
+  /// Controls whether the actual working tree on the filesystem is recursively removed.
+  ///
+  /// Defaults to `false`.
+  pub working_tree: Option<bool>,
+}
+
+impl From<WorktreePruneOptions> for git2::WorktreePruneOptions {
+  fn from(value: WorktreePruneOptions) -> Self {
+    let mut git2_worktree_prune_options = git2::WorktreePruneOptions::new();
+    value.valid.map(|_valid| git2_worktree_prune_options.valid(_valid));
```

Comment:
```suggestion
    if let Some(valid) = value.valid {
      git2_worktree_prune_options.valid(valod);
    }
```
`.map` is is not a suitable operator to perform mutable operations. Can we change this line?